File: Recursion_and_Backtracking_pepcoding_level_1/1.Recursion_Intro/7.tower_of_hanoi.js
const toh = (n, src, des, hlp) => {
    if (n === 0) return;

    toh(n - 1, src, hlp, des); // move n-1 from src to hlp
    console.log(n + " [" + src + " -> " + des + "]");
    toh(n - 1, hlp, des, src); // bring back n-1 from hlp to des
};

toh(3, "A", "B", "C"); 

File: Recursion_and_Backtracking_pepcoding_level_1/1.Recursion_Intro/7.tower_of_hanoi.rb
def toh n, src, des, hlp
  return if n == 0

  toh(n-1, src, hlp, des);
  puts "#{n} [#{src} -> #{des}]"
  toh(n-1, hlp, des, src)
end

toh 3, "A", "B", "C"
